services-directions-refresh/src/main/java/com/mapbox/api/directionsrefresh/v1/MapboxDirectionsRefresh.java

```diff
@@ -42,6 +45,25 @@ protected MapboxDirectionsRefresh() {
     );
   }
 
+  @Override
+  protected synchronized OkHttpClient getOkHttpClient() {
+    if (okHttpClient == null) {
```

Comment:
Would it be worth it to put this in `MapboxService` so we can use it from all services?

Reply:
Yeah, that's correct. I also thought about this but would mean a bigger refactor because there are already some other services that accept `Interceptor`s.

Let's ticket this out and revisit later.

Reply:
Follow up ticket 👉 https://github.com/mapbox/mapbox-java/issues/1041